diff --git a/http_parser.h b/http_parser.h
index c03ec05..747f47b 100644
--- a/http_parser.h
+++ b/http_parser.h
@@ -24,23 +24,8 @@
 extern "C" {
 #endif

-
 #include <sys/types.h>
-#if defined(_WIN32) && !defined(__MINGW32__)
-typedef __int8 int8_t;
-typedef unsigned __int8 uint8_t;
-typedef __int16 int16_t;
-typedef unsigned __int16 uint16_t;
-typedef __int32 int32_t;
-typedef unsigned __int32 uint32_t;
-typedef __int64 int64_t;
-typedef unsigned __int64 uint64_t;
-
-typedef unsigned int size_t;
-typedef int ssize_t;
-#else
 #include <stdint.h>
-#endif

 /* Compile with -DHTTP_PARSER_STRICT=0 to make less checks, but run
  * faster
@@ -140,9 +132,15 @@
 };


+#if defined(_WIN32) && !defined(__MINGW32__)
+__declspec(dllexport)
+#endif
 void http_parser_init(http_parser *parser, enum http_parser_type type);


+#if defined(_WIN32) && !defined(__MINGW32__)
+__declspec(dllexport)
+#endif
 size_t http_parser_execute(http_parser *parser,
                            const http_parser_settings *settings,
                            const char *data,
